### DISCUSSION TITLE
Model simplification

### DISCUSSION THREAD

**Discussion by okenk** [pfmc-assessments/canary_2023#31] [General]
The 2015 canary assessment was an extremely complex assessment model with many non-standard parameters for spatial recruitment deviations and unconfined selectivity functions. We may want to simplify the model this year to 1) overcome long run times, which will improve exploration of various assumptions (such as how to model the lack of old females) and 2) avoid overparameterization, since overparameterization increases out of sample predictive error.

[This paper](https://academic.oup.com/icesjms/article/76/6/1477/5475859) focused on overcoming long run times for bayesian stock assessments and included the 2015 canary assessment as a case study. It found that fixing some of the recruitment and selectivity parameters led to a 144-fold increase in MCMC efficiency with <1% changes to terminal year estimates of depletion, OFL, and SSB. While we are not planning on running a bayesian assessment, I think major gains in optimization efficiency for MLE would also be expected.

We may want to consider making some of these similar simplifications for our baseline model.

**Comment by kellijohnson-NOAA** [comment]
@okenk let me know if you are interested in running Cole's regularization procedure as a diagnostic. I did it for lingcod and have done it on the sablefish model. It was extremely helpful in identifying which parameters are correlated and unidentifiable given the data.

**Comment by brianlangseth-NOAA** [comment]
Candidate ways for simplification

- [ ] Reduce number of fleets:
-Triennial survey likely adds very little to the analysis but adds a separate fleet
-Survey catches (and thus comps) are included and perhaps do not need to be; keep as survey fleets
-WA commercial NTWL has very few age/length samples and of those collected are similar to WA TWL
- [ ] Fix some selectivity parameters, dont need to estimate all 6 for double normal
- [ ] Reduce comps
-Commercial age comps have separate surface age and break and burn compositions, sometime both for the same year (see #11)
- [ ] Remove spatial allocation of recruitment devs

**Reply by okenk** [reply]
This looks good. I would also add:

- [ ] eliminate the extra spatial rec devs.

**Reply by brianlangseth-NOAA** [reply]
Edited within the above

**Comment by brianlangseth-NOAA** [comment]
I ran the 2015 model within the 'post-SSC base model' folder from the archives. Currently it is set up to read from the par file so naturally run time is fast, so I changed that, and explored with and without the hessian. It runs faster than I expected. The benefit to any simplifications then is primarily to avoid overparameterization.

Run times are as follows:
33 sec - as is without a hessian
22 min 5 sec - as is with a hessian
9 min 10 sec - not reading from the par file without a hessian
29 min 27 sec - not reading from the par file with a hessian
